### ISSUE TITLE
What you can do when you get Max decoder steps reached

### ISSUE THREAD

**Issue by ErfolgreichCharismatisch** [NVIDIA/tacotron2#504]
Whenever you get the max decoder steps reached in inference, your audio text pairs have errors. You have to train from scratch again with good pairs.

**Comment by Fennecai** [comment]
might be related to my issue #502 since I get the same error iirc- i still don't know the answer either though, sorry.

**Comment by CookiePPP** [comment]
https://github.com/NVIDIA/tacotron2/blob/185cd24e046cc1304b4f8e564734d2498c6e2e6f/hparams.py#L59

You can change the max number of steps, however the model can have issues above 1000.

Normally this statement means you have input too much text to the model for inference or your model is not trained well.

**Comment by ErfolgreichCharismatisch** [comment]
> your model is not trained well.

That's what I am referring to.

**Comment by CookiePPP** [comment]
1. Ensure you have a large amount of data (30+ minutes)
2. Use a pretrained model as a base to [`--warm_start`](https://github.com/NVIDIA/tacotron2#training-using-a-pre-trained-model) from
3. Ensure all transcripts in your dataset match the audio perfectly
4. Remove any files with background noise
5. Perform inference in batches and filter out spectrograms with poor alignments or use beam search/greedy search style inference with chunks of frames.

If things still fail.

6. Decrease hop_length if your speaker speaks abnormally fast and retrain the vocoder + tacotron from scratch on a large dataset before transfer-learning OR
7. Use a different model that uses duration based alignment

**Comment by Fennecai** [comment]
> 1. Ensure you have a large amount of data (30+ minutes)
> 2. Use a pretrained model as a base to `--warm_start` from
> 3. Ensure all transcripts in your dataset match the audio perfectly
> 4. Remove any files with background noise
> 5. Perform inference in batches and filter out spectrograms with poor alignments or use beam search/greedy search style inference with chunks of frames.
> 
> If things still fail.
> 
> 1. Decrease hop_length if your speaker speaks abnormally fast and retrain the vocoder + tacotron from scratch on a large dataset before transfer-learning OR
> 2. Use a different model that uses duration based alignment

this will be helpful for my issue too I'm sure but... I'm honestly new to this and have no clue what some of this means. specifically:

2:  where do you put --warm_start and such? how do you use that?
5: can you describe this again on a complete-idiot's level?

**Comment by CookiePPP** [comment]
@Fennecai 
https://github.com/NVIDIA/tacotron2#training-using-a-pre-trained-model
Example usage is on the README.

---

> 5: can you describe this again on a complete-idiot's level?

Not really. This is for the people who can write the inference code more than people who are just trying to use it. But it's a method of filtering out poor results + instability that works quite well.

**Comment by ErfolgreichCharismatisch** [comment]
> that works quite well.

I have to agree. It's good to know that if there's an error, it's not in the original model so the area where problems arise is very limited and under your control.
I learnt it the hard way that a fastidiously fostered dataset is the be-all and end-all.

**Comment by toanil315** [comment]
@CookiePPP hi, can you tell me what is the maximum duration of a file .wav in dataset. (because some file in my dataset > 11.5s and all file < 12s)
Thanks

**Comment by CookiePPP** [comment]
@toanil315 
There is no limit to the maximum duration of your training/validation audio files. The `max_decoder_steps` is used when generating new audio using Tacotron2 + WaveGlow with new text.

---

In fact, training on longer audio files will increase the `max_decoder_steps` that you can use safely with the model.
(a dataset with maximum duration of 5s lets you use 1250 `max_decoder_steps`, a dataset with maximum duration of 10s lets you use 2000 `max_decoder_steps`)
![image](https://user-images.githubusercontent.com/42448678/160491013-107806d7-8752-4587-854a-59f3e1ab51a3.png)

**Comment by toanil315** [comment]
@CookiePPP thanks for reply, have good day

**Comment by toanil315** [comment]
@CookiePPP when i train from scratch, should i change learning rate with recipe 0.001 * (batch_size/256)**0.5
because with my hardware i'm using, i'll use with batch_size = 16

**Comment by toanil315** [comment]
@CookiePPP hi, this is my alignment after 38k iterations 
![image](https://user-images.githubusercontent.com/58522459/161245225-004689b5-3dc8-43b2-9ddd-baa8337466d7.png)
i'm traning with batch_size = 8 and learning_rate = 1.8e-4 (with recipe: 0.001 * (batch_size/256)**0.5)
this seem model can't learn
should i wait for some iterations or retrain it?
this is my dataset [https://drive.google.com/file/d/1QbvrLPjSQX6vrU2Zor8zOXIqg4xM9m4h/view?usp=sharing](https://drive.google.com/file/d/1QbvrLPjSQX6vrU2Zor8zOXIqg4xM9m4h/view?usp=sharing)
thanks
